voicevox_engine/cancellable_engine.py

```diff
@@ -24,50 +25,75 @@ class CancellableEngine:
 
     Attributes
     ----------
-    watch_con_list: List[Tuple[Request, Process]]
+    watch_con_list: list[tuple[Request, Process]]
         Requestは接続の監視に使用され、Processは通信切断時のプロセスキルに使用される
         クライアントから接続があるとListにTupleが追加される
         接続が切断、もしくは音声合成が終了すると削除される
-    procs_and_cons: queue.Queue[Tuple[Process, Connection]]
+    procs_and_cons: queue.Queue[tuple[Process, ConnectionBase]]
         音声合成の準備が終わっているプロセスのList
         （音声合成中のプロセスは入っていない）
     """
 
-    def __init__(self, args: argparse.Namespace) -> None:
+    use_gpu: bool
+    voicelib_dirs: list[Path] | None
+    voicevox_dir: Path | None
+    runtime_dirs: list[Path] | None
+    cpu_num_threads: int | None
+    enable_mock: bool
+
+    watch_con_list: list[tuple[Request, Process]]
+    procs_and_cons: queue.Queue[tuple[Process, ConnectionBase]]
```

Comment:
インスタンス変数の型をここに書くのは他のクラスではやってないことだった記憶があって、方針どうしようかちょっと迷ってます。
多分書かなくても`self.use_gpu`などは正しくboolだと認識されると思いますが、どっちの方が良いかなどPEP等で議論があったりしそうでしょうか 👀 

どの言語でもよく出てくる、「型宣言の二度手間」ｖｓ「宣言することで安全寄りになる」の一長一短だと感じてます 😇 
個人的にはどちらか一方に寄せるべきだと思っていて、他のすべてはこれを書いていないのであれば合わせるべきだと思っています。
逆にこうするのであれば、よほどのことがない限りほぼ全てのクラスでこの形にするべきだろうなぁと。
linterがあると嬉しいですね！

Reply:
なるほどです、たしかに冗長な気がしますね...

dataclassやPydanticのBaseModelでは`__init__`関数をほぼ書かないので考える必要がないですが、素のクラスを書くときには問題になりそうです。

今回クラス側にインスタンス変数の型を書くようにしたのは、以下のように最低限の型ヒントでコードを追加すると、一部の代入コードでだけ型ヒントが付くのがバランスが悪いと思ったからです。

```python
self.use_gpu = use_gpu
self.voicelib_dirs = voicelib_dirs
self.voicevox_dir = voicevox_dir
self.runtime_dirs = runtime_dirs
self.cpu_num_threads = cpu_num_threads
self.enable_mock = enable_mock

self.watch_con_list: list[tuple[Request, Process]] = []
self.procs_and_cons: queue.Queue[tuple[Process, ConnectionBase]] = queue.Queue()
```

> 書かなくてもself.use_gpuなどは正しくboolだと認識される

少なくとも単純な代入であれば、たしかに正しく認識されます。

> PEP等で議論

変数の型ヒント構文に関する[PEP 526](https://peps.python.org/pep-0526/#class-and-instance-variable-annotations)の[Rejected/Postponed Proposals](https://peps.python.org/pep-0526/#rejected-postponed-proposals)に関連しそうな項目がありました。

> **Allow instance variable annotations only in methods**: The problem is that many `__init__` methods do a lot of things besides initializing instance variables, and it would be harder (for a human) to find all the instance variable annotations. And sometimes `__init__` is factored into more helper methods so it’s even harder to chase them down. Putting the instance variable annotations together in the class makes it easier to find them, and helps a first-time reader of the code.

関数内でのみインスタンス変数の型ヒント（型アノテーション）を許可するべきか、という点について否定的みたいです。

`__init__`関数が複雑な場合、人間が型を判別するときにロジックを追う必要があるので、クラスの先頭を見るだけで型がわかるように、関数内だけでなくクラス側にも型ヒントがあった方がいい、という感じでしょうか。

個人的には、そもそも`__init__`関数を複雑にしたくない気もしますが、dataclassやBaseModelだけで書くというのも難しいと思いました。

Reply:
@aoirint あ、ここどうしましょ？

他のすべてのクラスをこの形に書き換えるか、このクラスを他に合わせるかのどちらかだと思います。
そんなに強い意見じゃないなら、一旦消しちゃう形が丸いのかなぁと思いました。

ペンディングになっているのが一番良くないので、特にご意見なければこっちで勝手に消させて頂こうかなと思います 🙇

Reply:
方針助かります。そんなに強い意見ではないので、このPRではいったん消しちゃいます！

クラス側に記述したインスタンス変数の型ヒントはなくして、`watch_con_list`だけ`__init__`内の型ヒントが必要なので、以前の形式に戻すコミットを追加しました。

Reply:
了解しました！